iina/Logger.swift

```diff
@@ -8,10 +8,20 @@
 
 import Foundation
 
-class Logger {
+struct Logger {
 
-  enum LogLevel: Int, Comparable, CustomStringConvertible {
-    static func < (lhs: LogLevel, rhs: LogLevel) -> Bool {
+  struct Subsystem: RawRepresentable {
```

Comment:
I'd prefer if this was an enum; is there any particular reason why you went with a struct? Plus, I'd add an `init(_ rawValue: String)` constructor.

Reply:
There's nothing to enumerate; each component should create their own subsystem and their names are not fixed. I think it's more like a `NSImage.Name` or `NSNib.Name`.

Reply:
> I think it's more like a NSImage.Name or NSNib.Name.

That's the API I was aiming for. The reason why this isn't a struct as those are is because LogLevel cannot have an arbitrary value as those do.